addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/MongoDbInjectionScanRule.java

```diff
@@ -362,92 +341,85 @@ public void scan(HttpMessage msg, String param, String value) {
-                    sleepValueToInj =
-                            SLEEP_INJECTION[index][0].replaceFirst(
-                                    INSERT_SHORT_TIME, Integer.toString(timeShort));
-                    phase = INSERT_SHORT_TIME;
-                    setParameter(msgInjAttack, param, sleepValueToInj);
-                    sendAndReceive(msgInjAttack, false);
-                    LOGGER.debug(
-                            "Trying for a longer time with the value: {}",
-                            SLEEP_INJECTION[index][1]);
-                    if (msgInjAttack.getTimeElapsedMillis() >= aveRtt + SHORT_THRESHOLD) {
-                        phase = INSERT_LONG_TIME;
-                        sleepValueToInj =
-                                SLEEP_INJECTION[index][1].replaceFirst(
-                                        INSERT_LONG_TIME, Integer.toString(timeLong));
-                        msgInjAttack = getNewMsg();
-                        setParameter(msgInjAttack, param, sleepValueToInj);
-                        sendAndReceive(msgInjAttack, false);
-                        if (msgInjAttack.getTimeElapsedMillis() >= aveRtt + LONG_THRESHOLD) {
-                            newAlert()
-                                    .setConfidence(Alert.CONFIDENCE_HIGH)
-                                    .setParam(param)
-                                    .setAttack(sleepValueToInj)
-                                    .setOtherInfo(getExtraInfo(SLEEP_ATTACK))
-                                    .setMessage(msgInjAttack)
-                                    .raise();
-                            isBingo = true;
-                            break;
-                        }
+                    try {
+                        // use TimingUtils to detect a response to sleep payloads
+                        isInjectable =
+                                TimingUtils.checkTimingDependence(
+                                        MAX_DELAY,
+                                        BLIND_REQUEST_LIMIT,
+                                        BLIND_SECONDS_LIMIT,
+                                        requestSender,
+                                        TIME_CORRELATION_ERROR_RANGE,
+                                        TIME_SLOPE_ERROR_RANGE);
+                    } catch (SocketException ex) {
+                        LOGGER.debug(
+                                "Caught {} {} when accessing: {}.\n The target may have replied with a poorly formed redirect due to our input.",
+                                ex.getClass().getName(),
+                                ex.getMessage(),
+                                message.get().getRequestHeader().getURI());
+                        continue; // Something went wrong, move to next blind iteration
                     }
-                    if (hadTimeout) {
+
+                    if (isInjectable) {
+                        // We Found IT!
+                        // First do logging
```

Comment:
I'm fine with the found it comment (I guess) but the logging comment probably isn't needed, LOGGER.debug makes it pretty clear.